### PR TITLE
MAINT: Use Black for formatting

### LINKED ISSUE #1143: Use Black for formatting
Following the change in MNE-Python I think we should start using black in mne-bids as well.

- https://github.com/mne-tools/mne-python/pull/11667

### PR DESCRIPTION
**Pull request by sappelhoff**
PR Description
--------------

fixes #1143 

Let's see how this goes.

Merge checklist
---------------

Maintainer, please confirm the following before merging.
If applicable:

- [ ] All comments are resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [ ] PR title starts with [MRG]
- [ ] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/main/doc/whats_new.rst) is updated
- [ ] New contributors have been added to [CITATION.cff](https://github.com/mne-tools/mne-bids/blob/main/CITATION.cff)
- [ ] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

**Comment by hoechenberger** [comment]
What's with the setuptools pinning..?

**Comment by sappelhoff** [comment]
I'm getting:

```
/home/stefanappelhoff/miniconda3/envs/mne_bids/lib/python3.10/site-packages/setuptools/dist.py:543: UserWarning: The version specified ('"0.13.dev0"') is an invalid version, this may not work as expected with newer versions of setuptools, pip, and PyPI. Please see PEP 440 for more details.
  warnings.warn(
Traceback (most recent call last):
  File "/home/stefanappelhoff/Desktop/bids/mne-bids/setup.py", line 24, in <module>
    setup(
  File "/home/stefanappelhoff/miniconda3/envs/mne_bids/lib/python3.10/site-packages/setuptools/__init__.py", line 87, in setup
    return distutils.core.setup(**attrs)
  File "/home/stefanappelhoff/miniconda3/envs/mne_bids/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 185, in setup
    return run_commands(dist)
  File "/home/stefanappelhoff/miniconda3/envs/mne_bids/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
    dist.run_commands()
  File "/home/stefanappelhoff/miniconda3/envs/mne_bids/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
    self.run_command(cmd)
  File "/home/stefanappelhoff/miniconda3/envs/mne_bids/lib/python3.10/site-packages/setuptools/dist.py", line 1208, in run_command
    super().run_command(command)
  File "/home/stefanappelhoff/miniconda3/envs/mne_bids/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
    cmd_obj.run()
  File "/home/stefanappelhoff/miniconda3/envs/mne_bids/lib/python3.10/site-packages/setuptools/command/sdist.py", line 48, in run
    self.run_command('egg_info')
  File "/home/stefanappelhoff/miniconda3/envs/mne_bids/lib/python3.10/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command
    self.distribution.run_command(command)
  File "/home/stefanappelhoff/miniconda3/envs/mne_bids/lib/python3.10/site-packages/setuptools/dist.py", line 1208, in run_command
    super().run_command(command)
  File "/home/stefanappelhoff/miniconda3/envs/mne_bids/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 987, in run_command
    cmd_obj.ensure_finalized()
  File "/home/stefanappelhoff/miniconda3/envs/mne_bids/lib/python3.10/site-packages/setuptools/_distutils/cmd.py", line 111, in ensure_finalized
    self.finalize_options()
  File "/home/stefanappelhoff/miniconda3/envs/mne_bids/lib/python3.10/site-packages/setuptools/command/egg_info.py", line 219, in finalize_options
    parsed_version = parse_version(self.egg_version)
  File "/home/stefanappelhoff/miniconda3/envs/mne_bids/lib/python3.10/site-packages/pkg_resources/_vendor/packaging/version.py", line 266, in __init__
    raise InvalidVersion(f"Invalid version: '{version}'")
pkg_resources.extern.packaging.version.InvalidVersion: Invalid version: '-0.13.dev0-'
```

**Comment by hoechenberger** [comment]
Our version naming doesn't conform to PEP-440, we should fix that

**Comment by sappelhoff** [comment]
Agreed -- I don't see how we are violating the rules though: https://peps.python.org/pep-0440/#developmental-releases

**Comment by hoechenberger** [comment]
> '-0.13.dev0-'

the dashes maybe..?

**Comment by hoechenberger** [comment]
Maybe @larsoner can help switch versioning over to setuptools-scm

**Comment by sappelhoff** [comment]
> the dashes maybe..?

they are inserted by the tool, our version seems fine:

https://github.com/mne-tools/mne-bids/blob/a19b20f0952830ab731a41e10676f4e47e648316/mne_bids/__init__.py#L3

**Comment by sappelhoff** [comment]
Okay I think I caught it. It was a silly mistake in parsing the version, see:

https://github.com/mne-tools/mne-bids/blob/a19b20f0952830ab731a41e10676f4e47e648316/setup.py#L17

so you were right that something was fishy with the dashes, thanks!

**Comment by hoechenberger** [comment]
Glad you figured it out!

Still, maybe we should try switching to setuptools-scm like we did for MNE-Python and MNE-BIDS-Pipeline. Seems so much more convenient and can make the release process easier.

**Review by hoechenberger — APPROVED** [review]

File: .git-blame-ignore-revs [hunk 1]
```diff
@@ -0,0 +1 @@
+58d9cf0b3a916af3e48fbb63b85b699c998c7f7a
```
Comment (larsoner) [inline .git-blame-ignore-revs:1]: FYI a rebase and merge will change the relevant commit number

Comment (larsoner) [reply .git-blame-ignore-revs:1]: (at least it will assuming it actually does a rebase... if it's already up to date maybe it won't?)

Comment (sappelhoff) [reply .git-blame-ignore-revs:1]: oh damn, so I should create a merge commit? Thanks!

**Pull request merged**